Add Lando Adminer service, fixes #359.

tormi opened the pull request:
*Link to ticket: #359*

*Changes proposed in this PR:*
- Add the [Adminer](https://www.adminer.org/) Lando service based on the official [Adminer Docker image](https://hub.docker.com/_/adminer). We have several projects where Adminer is added via https://github.com/dehy/docker-adminer and the current implementation unfortunately causes extra build time because of hackish setup. Let's start using the official image instead with proper implementation. 

*How to test:*
1. Uncomment Adminer service & proxy address at `.lando.yml` and rebuild Lando via `lando rebuild -y`.
2. Validate that http://adminer.drupal-project.lndo.site address is green and you can use it (no credentials needed).

*Link to feature environment:*
- http://adminer.drupal-project.lndo.site

misterjoonas reviewed (approved):
Works nicely! No need for Sequel Pro anymore 👍

tormi commented:
Port `8080` is probably not the best choice here, because it's one of the `proxyHttpFallbacks` in [Lando](https://github.com/lando/core-next/blob/main/plugins/proxy/index.js#L30): 

`proxyHttpFallbacks: ['8000', '8080', '8888', '8008'],`

tormi commented:
> In the past we have added a [dehy/adminer](https://github.com/dehy/docker-adminer) image based setup to several projects, but with the latest Lando updates, it’s starting to have issues with long scans, status, and also login.

I switched to [dehy/adminer](https://github.com/dehy/docker-adminer) because I resolved the login issue and `dehy/adminer` setup is even smaller in size than official one.

hkirsman commented:
I've been including phpmyadmin in active state in all of my projects from the day I started using Lando. I'd add also adminer as enabled in the configs. You never know when you need it and if it has minimal load on the system, then why not?

hkirsman commented on .lando/adminer.php:
```diff
@@ -0,0 +1,31 @@
+    public function credentials()
+    {
+      // Server, username and password for connecting to database.
+      return array('database', 'drupal10', 'drupal10');
```
Wondering if DB_NAME_DRUPAL etc env variables should be in .env file instead them being .lando.yml This way we could use the env here.

tormi replied:
Unfortunately, current `adminer` service doesn't support envvars. Tried with these changes:

`.lando.yml`:

```yml
services:
  adminer:
    type: compose
    services:
      image: dehy/adminer
      command: /bin/s6-svscan /etc/services.d
      environment:
        DB_USER_DRUPAL: drupal10
        DB_PASS_DRUPAL: drupal10
        DB_HOST_DRUPAL: database
    run_as_root:
      - /app/.lando/adminer.sh
 ```
 
 `adminer.php`:
 
 ```php
    public function credentials()
    {
      // Server, username and password for connecting to database.
      return array(getenv('DB_HOST_DRUPAL'), getenv('DB_USER_DRUPAL'), getenv('DB_PASS_DRUPAL'));
    }
```

tormi replied:
But it's easy to update as is during Drupal major upgrade.

hkirsman commented on .lando/adminer.sh:
```diff
@@ -10,3 +10,23 @@ find . -type f -name 'adminer-*' -print0 | xargs -0 -I{} cp -f {} adminer_librar
 chmod -R +x /var/www/html
+
+#
+# Allow PHP-FPM to read environment variables.
```
@tormi I found a way to add env variables.

tormi replied:
I would add error handling and also use the `enable_clear_env_option()` naming to be more descriptive.

hkirsman commented on .lando/adminer.sh:
```diff
@@ -1,8 +1,8 @@
-# Copy custom adminer.php from .lando folder to webroot.
-cp -f /app/.lando/adminer.php /var/www/html/adminer.php
+# Link custom adminer.php from .lando folder to webroot.
+ln -sf /app/.lando/adminer.php /var/www/html/
```
@tormi I think it's better to link so for every change would wouldn't need to restart/rebuild Lando:

![image](https://github.com/wunderio/drupal-project/assets/492375/598da20c-45a9-4d8f-b98d-57053e7ff8b2)

tormi commented:
Good stuff, thanks @hkirsman! Should we move remaining `appserver` envvars to `.env` file as well?

artursv commented on .lando.yml:
```diff
@@ -44,6 +44,13 @@ tooling:
 
 services:
+  adminer:
+    type: compose
```
Compose service is deprecated in the latest versions of Lando. Is it maybe worth looking into some more future proof solutions? https://docs.lando.dev/compose/

tormi replied:
Thanks, @artursv! Good catch!

tormi replied:
@artursv, here: https://github.com/wunderio/drupal-project/pull/371. Thanks again for pointing out!
